### ISSUE TITLE
Problems creating a new exchange type and resource type in testocp

### ISSUE THREAD

**Issue by bhaugen** [FreedomCoop/valuenetwork#218]
@bum2  Here are the steps so you can reproduce them:
* Started at https://testocp.freedomcoop.eu/work/agent/96/
* clicked on https://testocp.freedomcoop.eu/work/agent/96/exchanges/
* clicked on the Show link to create a new exchange type
* Selected the Fair Economy tab
* Expanded the Fair rent item
* Clicked the fair rent Charge resources radio button
* Expanded Material/Spaces/Living space, where it looks like my new Resource Type from my last test, "Daily room rent" did get saved, although the new exchange type did not get saved.
* Clicked the button to create a new Resource Type.
* Created the new Resource Type "Monthly room package".
* Clicked the Save button on the modal form.
* Landed back on https://testocp.freedomcoop.eu/work/agent/96/exchanges/
   * with a banner saying "The new resource type request has some errors, click 'SHOW' to see them."
   * clicked SHOW
   * see a message "Parent resource type: This field is required." but my new resource type is hidden, and I don't know why a parent resource type is required, or what to select for a parent.

**Comment by bhaugen** [comment]
P.S. my new exchange type appears here,, too:
https://testocp.freedomcoop.eu/accounting/change-exchange-type/43/

**Comment by bhaugen** [comment]
Some suggestions for improvement, more to come:
* don't do a page reload when saving the new resource type. Save and close the modal dialog ajaxy. Keep the underlying page, with all of my previous selections, intact.

**Comment by bhaugen** [comment]
Retesting:
* when selecting a use case:
```
...choosing a use case:
  fair Sell  
  fair Buy  
  fair Rent  
  fair rent Pay resources <  
* fair rent Charge resources <  #I picked this one 
    fair Monthly Living space <  
```
    * I got a message on the right saying "...you created this type: Edit " [button]
    * I did not create that type, I created the one under it, "Monthly Living space" 
        * (minor problem)
* Same result as described above. Page reload, all my selections disappeared, no new exchange type. I was essentially starting over, not knowing what had happened (except this time I knew what had happened, but it would surprise the hell out of anybody else).

**Comment by bum2** [comment]
you are using your superuser account, and so you can edit any type (like me)... with a normal user you only can edit the types you've created..

**Comment by bhaugen** [comment]
> you are using your superuser account, and so you can edit any type (like me)..

Got it. As I said above, minor anyway. I wouldn't have raised an issue about that one.

**Comment by bhaugen** [comment]
Note: I edited the comment two above this one. Wasn't clear before that I had saved another new resource type, duplicating the whole sequence from the first post on this issue.

**Comment by bhaugen** [comment]
Screenshot:
![screenshot from 2017-03-09 16 24 39](https://cloud.githubusercontent.com/assets/117439/23773492/161abfce-04e5-11e7-8954-6050c31f4f10.png)

**Comment by bhaugen** [comment]
One more way you could improve the situation, I think easier than my [comment upthread](https://github.com/FreedomCoop/valuenetwork/issues/218#issuecomment-284379583), would be to separate the creation of new resource types from the creation of a new exchange type. That would avoid a page reload losing all of your selections in general-app tree climbdowns, and make it clear what happened with the new resource type. We'll need a way for projects to create new resource types anyway. You could just add a link to the resource type creation here, instead of doing it 3/4 of the way through the exchange type creation and losing the thread of selections.

**Comment by bum2** [comment]
the selections are lost because i've placed a js to clear the fields, meanwhile, because the behavior when there are selections onload it's not yet working good... was a fast and bad solution to start using the app without js errors... I understand is very important to keep the selections, and asap i will try to debug the js issues and let the fields filled when they should.
Also the behaviors in that page should be better via ajax in separate forms as you said (now is all the same form, reloading page). What we have now was the fastest way i found to start doing the things, sometimes with rude codes which have lot of room for improvement.
To improve the page, is pending: 
- let the selections in the fields onload and run the js on them
- a splitting of forms behaviors, validation, save and refresh via ajax
- load the trees and branches on demand via ajax
On any of this pending tasks or any of the todos in the pad, any help is very welcome!

I thought initially to have a page only for resource-types, but later i tried to merge that into the Resources page and finally i placed it in the Exchanges page as a modal. The same modal can be now placed in the resources page or wherever is needed the 'new resource type' creation.

**Comment by bhaugen** [comment]
> What we have now was the fastest way i found to start doing the things, sometimes with rude codes which have lot of room for improvement.

I totally understand.

**Comment by fosterlynn** [comment]
>On any of this pending tasks or any of the todos in the pad, any help is very welcome!

@bum2 My advice is to acknowledge right now that your release is not going to be ready for the training at the hackathon, no matter who helps you.  That will give you time to regroup around the training and do some preparation to train on the code in master.  You will have to explain that this is older (and possibly incomplete) code, there is a new release coming, and you could explain the differences.  It will be fine.  It will be much better than trying to train on code that isn't ready.

If you wait until the last minute, it will be a lot harder for whoever is giving the training to prepare.  And if you use master, you will be able to give people a taste of how they can actually manage their projects in OCP, even if it is not your longer term vision.

The hackathon itself can proceed on a different code base than the training.  If it is better on your branch, then you all can make that decision.  Coding can start from anywhere.

**Comment by bhaugen** [comment]
I have to agree with Lynn here. I don't think there is much I can do in the next two days that will make much difference, and I have a bunch of other things I need to do as well. I'd need to study your existing code and by then, the time is gone.

Then let's see what everybody wants to do in the hackathon, and make some decisions for the future.

**Comment by fosterlynn** [comment]
@bum2 by the way, I totally understand how you feel about it.  I have done that same thing in the past, more than once, just trying to crunch it through even when the odds are low.  It is pretty hard to step back in the middle of a deep coding focus to take stock, and look at what would have the best consequences for the whole process.

Also, if you make this decision in time, this would let you relax a little and enjoy doing the hackathon work, and connecting with those people, instead of having to keep crunching on this.  You will be the person closest to the code, so your full involvement in the hackathon work would be useful.
